### ISSUE TITLE
make public address mandatory for stake pool operators

### ISSUE THREAD

**Issue by laplasz** [input-output-hk/jormungandr#1537]
**Is your feature request related to a problem/context ? Please describe if applicable.**
Node operators starts not to setting up public_address parameter making the node more stable since the node is not discoverable so - less and less incomming request the node should deal with. But that is againts the p2p/decentralisation concepts in my point of you. it would end up the trusted peers would be only the public peers available in the network to sync with.

**Describe the solution you'd like**
exit with error if public_address is not set.

**Comment by consuman** [comment]
I don't think this will lead to everyone not setting a public IP. We do it right now, because it's much more stable, but as soon as the network goes back to normal conditions, I'll be happy to set it again.

**Comment by jalora8** [comment]
Yeah not sure if this is a valid request as it's a short term bandage for a problem that is already identified and is being addressed.

See #1531, #1521, #1459.

**Comment by laplasz** [comment]
yes - this solves an known issue as a workaround, but ideal case the staking pool should have a public address configured - as this is a [requirement](https://iohk.zendesk.com/hc/en-us/articles/360039418834-Stake-Pool-System-Requirements)
Or probably this is not a client side problem instead of a network one. I think in long term stake-pools without accessible port should be blacklisted.. or should not be a requirement..

**Comment by NicolasDP** [comment]
There are issues with network stability as reported by @jalora8 . Right now you see an advantage to have your node running in private setting. Once the network start to be more stable it will be the other way around.

**Comment by vincenthz** [comment]
tl;dr: on a perfectly well working network, you make yourself more likely to be less competitive by not having public ip.

To develop a bit further, this is *not* a requirement to set a public ip (despite what the document says), however since it's important for a pool to get the latest information of the network (e.g. latest blocks), it is way more convenient to be reachable on the network as it allow random other peers to sync to you directly instead of using a second class private communication channel that will inevitably mean that you depends on some specific peer(s) to get the information you need to stay up-to-date.

Right now with the network woes and bugs, it may improve your own pool, but if everyone runs with this settings, this is making the network conditions really bad.

**Comment by consuman** [comment]
Thank you for your explaination @vincenthz!

We have updated our setup four days ago, now having 10 nodes running in parallel, all with public IP and ID set. For a two node setup it was hard to maintain stable last week, but choosing always the healthiest out of ten made it feasible to have these settings in place and support network stability this way.
